commercetools/client.go

```diff
@@ -98,6 +100,15 @@ func (qi QueryInput) toParams() (values url.Values) {
 		values.Set("offset", strconv.Itoa(qi.Offset))
 	}
 
+	// This is a string and not a bool because it would be hard to differentiate if user explicitly set value to "false"
+	// or it is just the default value. If it is the default value we do not want to pass this query param to all
+	// of the requests that use query input.
+	// withTotal can only be "true" or "false" otherwise malformed URL will be thrown.
+	// https://docs.commercetools.com/http-api#pagedqueryresult
+	if qi.WithTotal == "true" || qi.WithTotal == "false" {
```

Comment:
Should we use a pointer to bool instead?

Reply:
i usually use pointers to bools when i need a tribool

Reply:
yeap i agree and like it better. Changed